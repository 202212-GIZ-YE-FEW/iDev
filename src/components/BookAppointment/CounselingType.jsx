import { withTranslation } from "next-i18next";

import RadioGroup from "../ui/radiogroup/RadioGroup";
import RadioInputItem from "../ui/radiogroup/RadioInputItem";

const CounselingType = ({ t, name, counselingType, onChange }) => {
    return (
        <>
            <RadioGroup title={t("counselingTypeTitle")} as='button'>
                <RadioInputItem
                    id='individual'
                    name={name}
                    value='individual'
                    content={t("individualCounseling")}
                    checked={counselingType === "individual"}
                    onChange={(e) => onChange(e)}
                />
                <RadioInputItem
                    id='teen'
                    name={name}
                    value='teen'
                    content={t("teenCounseling")}
                    checked={counselingType === "teen"}
                    onChange={(e) => onChange(e)}
                />
            </RadioGroup>
        </>
    );
};

export default withTranslation("appointment")(CounselingType);
