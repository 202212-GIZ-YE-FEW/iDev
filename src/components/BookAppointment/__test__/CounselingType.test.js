import renderer from "react-test-renderer";

import CounselingType from "../CounselingType";

it("renders correctly", () => {
    const tree = renderer.create(<CounselingType />).toJSON();
    expect(tree).toMatchSnapshot();
});
