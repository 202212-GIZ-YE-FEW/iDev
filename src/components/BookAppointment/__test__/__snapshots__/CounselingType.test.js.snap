// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <p
    className="mb-4 p-2 text:lg md:text-2xl lg:text-3xl font-normal break-words text-black"
  >
    counselingTypeTitle
  </p>,
  <fieldset
    className="space-y-4"
  >
    <div
      className="text-9xl font-normal text-black"
    >
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="individual"
          onChange={[Function]}
          type="radio"
          value="individual"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="individual"
        >
          individualCounseling
        </label>
      </div>
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="teen"
          onChange={[Function]}
          type="radio"
          value="teen"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="teen"
        >
          teenCounseling
        </label>
      </div>
    </div>
  </fieldset>,
]
`;
